### PR TITLE
remove dependency of ocs-client-operator CRDs during manager setup

### PR DESCRIPTION
**Pull request by leelavg**
use the recently introduced CRD checks feature for removing dependency of ocs-client-operator CRDs by storagecluster controller.

### CODE REVIEW

**Comment by leelavg** [comment]
removal of the key is being done via https://github.com/red-hat-storage/ocs-client-operator/pull/260

**Comment by leelavg** [comment]
/hold

testing in progress

**Comment by rewantsoni** [comment]
LGTM

File: hack/install-ocs-operator.sh [hunk 1]
```diff
@@ -32,40 +32,16 @@ data:
...
-        if oc get cm ocs-client-operator-config -n openshift-storage; then
-            oc patch cm ocs-client-operator-config -n openshift-storage --type merge -p '{"data":{"DEPLOY_CSI":"false"}}'
-            sleep 2
-        fi
-    done
-}
-
-
 "$OPERATOR_SDK" run bundle "$ROOK_BUNDLE_FULL_IMAGE_NAME" --timeout=10m --security-context-config restricted -n "$INSTALL_NAMESPACE"
 "$OPERATOR_SDK" run bundle "$CSI_ADDONS_BUNDLE_FULL_IMAGE_NAME" --timeout=10m --security-context-config restricted -n "$INSTALL_NAMESPACE"
```
Comment (iamniting) [inline hack/install-ocs-operator.sh:36]: We don't even need CSI addons, It was being deployed because of the OCS-client-operator.

**Review by iamniting — CHANGES_REQUESTED** [review]

**Comment by leelavg** [comment]
/unhold

tested using a full custom d/s build based on this PR, extra fixes added after the review are
``` diff
diff --git a/controllers/util/predicates.go b/controllers/util/predicates.go
index 804647b7b..aaf5c1666 100644
--- a/controllers/util/predicates.go
+++ b/controllers/util/predicates.go
@@ -1,8 +1,10 @@
 package util
 
 import (
-       "github.com/go-logr/logr"
+       "fmt"
        "reflect"
+
+       "github.com/go-logr/logr"
        "sigs.k8s.io/controller-runtime/pkg/client"
 
        "sigs.k8s.io/controller-runtime/pkg/event"
@@ -83,13 +85,13 @@ func CrdCreateAndDeletePredicate(log *logr.Logger, crdName string, crdExists boo
        return predicate.Funcs{
                CreateFunc: func(_ event.CreateEvent) bool {
                        if !crdExists {
-                               log.Info("CustomResourceDefinition %s was Created.", crdName)
+                               log.Info(fmt.Sprintf("CustomResourceDefinition %s was Created.", crdName))
                        }
                        return !crdExists
                },
                DeleteFunc: func(_ event.DeleteEvent) bool {
                        if crdExists {
-                               log.Info("CustomResourceDefinition %s was Deleted.", crdName)
+                               log.Info(fmt.Sprintf("CustomResourceDefinition %s was Deleted.", crdName))
                        }
                        return crdExists
                },
```

File: hack/install-ocs-operator.sh (hunk 1 above)
Comment (leelavg) [reply hack/install-ocs-operator.sh:36]: done.

Comment (leelavg) [reply hack/install-ocs-operator.sh]: done.

**Comment by leelavg** [comment]
/cherrypick release-4.17
when the windows opens for 4.17 patch release.

File: hack/install-ocs-operator.sh [hunk 2]
```diff
@@ -39,7 +39,6 @@ EOF
 
 oc wait --timeout=5m --for condition=Available -n "$INSTALL_NAMESPACE" deployment \
     ceph-csi-controller-manager \
-    csi-addons-controller-manager \
```
Comment (iamniting) [inline hack/install-ocs-operator.sh:65]: This change should be part of the first commit.

**Review by iamniting — CHANGES_REQUESTED** [review]

File: hack/install-ocs-operator.sh (hunk 2 above)
Comment (leelavg) [reply hack/install-ocs-operator.sh:65]: I can change but don't want to re-trigger whole OCP and OCS deploy, could you override that?

Comment (iamniting) [reply hack/install-ocs-operator.sh:65]: I can do that not an issue.

Comment (leelavg) [reply hack/install-ocs-operator.sh:65]: done.

**Review by iamniting — APPROVED** [review]

**Comment by rewantsoni** [comment]
@leelavg I am not sure if we can backport this to 4.17 directly, the avail-crd feature is not backported

**Comment by leelavg** [comment]
Ack, will revisit if client-op in ci becomes bottleneck for backports to 4.17.

**Pull request merged**